metadata.yaml

```diff
@@ -0,0 +1,43 @@
+# Copyright 2023 Canonical Ltd.
+# See LICENSE file for licensing details.
+
+name: ranger-k8s
+ 
+display-name: Ranger K8s Operator
+
+summary: Kubernetes operator for Apache Ranger
+
+description: |
+ Apache Ranger is a framework to enable, monitor, and manage comprehensive data security.
+
+maintainers:
+  - Commercial Systems <jaas-crew@lists.canonical.com>
+
+source: https://github.com/canonical/ranger-k8s-operator
+docs: https://github.com/canonical/ranger-k8s-operator
+issues: https://github.com/canonical/ranger-k8s-operator/issues
+
+peers:
+  peer:
+    interface: ranger
+
+requires:
+  database:
+    interface: postgresql_client
+    limit: 1
+  nginx-route:
+    interface: nginx-route
+
+provides:
+  policy:
+    interface: ranger_client
+
+containers:
+  ranger:
+    resource: ranger-image
+
+resources:
+  ranger-image:
+    type: oci-image
+    description: OCI image for Apache Ranger
+    upstream-source: ghcr.io/canonical/ranger:2.4.0
```

Comment:
It's advisable to version the upstream source against a image digest instead of a tag. That's debatable and depends a lot on how the image is maintained though.

Reply:
Good to know! I have updated with this digest.